File: mem/src/main/java/me/wsj/mem/MemoryInfo.java
package me.wsj.mem;

import android.os.Debug;

public class MemoryInfo {
    public String procName;
    public AppMemory appMemory;
    public SystemMemory systemMemoryInfo;
    public String display;
    public int activityCount;

    public static class AppMemory {
        public long dalvikPss;//java占用内存大小
        public long nativePss;//前进程总私有已用内存大小
        public long totalPss;//当前进程总内存大小
        public Debug.MemoryInfo mMemoryInfo;
    }

    public static class SystemMemory {
        public long availMem;
        public boolean lowMemory;
        public long threshold;
        public long totalMem;
    }
}

File: outsider/src/main/java/me/wsj/apm/mem/MemoryInfo.kt
package me.wsj.apm.mem

import android.os.Debug

data class MemoryInfo(
    var procName: String? = null,
    var appMemory: AppMemory? = null,
    var systemMemoryInfo: SystemMemory? = null,
    var display: String? = null,
    var activityCount: Int = 0
)

data class AppMemory(
    //java占用内存大小
    var dalvikPss: Long = 0,
    //前进程总私有已用内存大小
    var nativePss: Long = 0,
    //当前进程总内存大小
    var totalPss: Long = 0,
    var mMemoryInfo: Debug.MemoryInfo? = null
)

data class SystemMemory(
    var availMem: Long = 0,
    var lowMemory: Boolean = false,
    var threshold: Long = 0,
    var totalMem: Long = 0
)